Update documentation on Verto 'relative-link' behaviour (fixes #504)

JackMorganNZ opened the pull request:
Also fixes an internal link.

Fixes #504.

hayleyavw commented on docs/source/author/writing_guide.rst:
```diff
@@ -227,47 +227,62 @@ destination of the link.
-it helps to have knowledge of the resulting URL path for a file.
-See the examples below:
+Links to pages within the website use the Django ``url`` template tag.
+In order to create an link, use the values that would be given to the template tag.
```
`create an link` -> `create a link`

hayleyavw commented:
Slightly confused, the relative-link template doesn't seem to include params?

hayleyavw commented on docs/source/author/writing_guide.rst:
```diff
@@ -227,47 +227,62 @@ destination of the link.
 .. code-block:: none
 
-  Please [contact us](about/index.html#contact).
+  [Little Red Riding Hood cards]('resources:resource' 'sorting-network-cards'?type=riding_hood)
```
pls clarify,  the template for relative-link doesn't inlcude params, is `type=riding_hood` just included as part of the path? in which case, how is this parsed (there are quotes around `sorting-network-cards`, what does this mean). Basically, what is this magic?

JackMorganNZ commented:
Addressed your comments, it was hard to expand on how the links worked without getting overly technical in the authors document (hence the warning message). 

The extra parameters are included in the `{{ link_query }}` in the template:

```html
<a href="{{ '{% url' }} {{ link_path }} {{ '%}' }}{{ link_query }}">{{ text }}</a>
```

hayleyavw commented on docs/source/author/writing_guide.rst:
```diff
@@ -266,7 +266,13 @@ Link to resource page with pre-selected options
+at the end.
+The Markdown processor Verto will read the link string and seperate everything
+before and after the ``?`` symbol, and will create a link using the Django URL tag
+with everything before symbol (the same way a normal internal link is created).
```
`before symbol` -> `before the symbol`
